middleware/http/spancloser.go

```diff
@@ -0,0 +1,23 @@
+package http
+
+import (
+	"io"
+	"time"
+
+	zipkin "github.com/openzipkin/zipkin-go"
+)
+
+type spanCloser struct {
+	io.ReadCloser
+	sp           zipkin.Span
+	traceEnabled bool
+}
+
+func (s *spanCloser) Close() (err error) {
+	if s.traceEnabled {
+		s.sp.Annotate(time.Now(), "Body Close")
```

Comment:
just food for thought, this difference between when headers are read and when body (if ever) is read leads to hard decisions like this. It is perfectly fine to have a annotation outside the span start/duration btw. the semantics of a client or server start/stop span.timestamp/duration are certainly grounds for clarity.. you can feel free to lead the charge with zipkin-go taking the modeling beyond status quo, similar to how is being fleshed out here https://docs.google.com/document/d/1yIo_zZvMyAS8J7bpCz85h-FISwU0Jr-Zu-POIvp7B9E/edit

Reply:
Luckily in Go it is an absolute must for a user to close the body of a successful response on a client call, even if not interested in reading / pulling all of the response body. This means we’re guaranteed this is called or the application leaks / does not release connections back to the connection pool, ... anyway.

If a user does not want the application span to encompass handling of the body they can use NewTransport in the standard http client and either manage their own application span or even omit an application span.

This all needs to be clarified in docs, examples, tests.